aws/defaults/defaults.go

```diff
@@ -154,6 +158,15 @@ func isLoopbackHost(host string) (bool, error) {
 	return true, nil
 }
 
+// isAllowedHost allows host to be loopback host,ECS container host 169.254.170.2
+// and EKS container host 169.254.170.23
+func isAllowedHost(host string) (bool, error) {
+	if host == ECSContainerHost || host == EKSContainerHost {
```

Comment:
this doesn't do DNS lookup like `isLoopbackHost` does. We need to preserve that functionality.

You'll likely want to basically turn `isLoopbackHost` into `isAllowedHost`, which handles the parse/DNS lookup as necessary, and then checks if it's one of the three allowed values.